drivers/cifutils.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python
+#
+# Copyright (C) Citrix Systems Inc.
+#
+# This program is free software; you can redistribute it and/or modify 
+# it under the terms of the GNU Lesser General Public License as published 
+# by the Free Software Foundation; version 2.1 only.
+#
+# This program is distributed in the hope that it will be useful, 
+# but WITHOUT ANY WARRANTY; without even the implied warranty of 
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+#
+# ISOSR: remote iso storage repository
+
+import SR, VDI, SRCommand, util
+import nfs
+import os, re
+import xs_errors
+
+
+def getCIFCredentials(dconf, session):
+    credentials={}
+    if dconf.has_key('username') \
+            and (dconf.has_key('cifspassword') or dconf.has_key('cifspassword_secret')):
```

Comment:
The SMB SR just uses password/password_secret so this needs to be overriddeble by the caller

Reply:
fixed